The typical datafile used for analysis in the BMPL is a .h5 file which uses a storage format called HDF5. To load at dataset into RAM memory for doing analysis, use the following code:

In [2]:
from load_hdf5 import load_hdf5
datafilename= 'Dataset4_wiretarget_1p5kV_1msstuff_0kAwire_density_10shots.h5'
data1=load_hdf5(datafilename,verbose=True)

All Groups Contained
discharge
discharge/dis_I
discharge/dis_I_raw
discharge/dis_V
discharge/dis_V_raw
isat probe
isat probe/isat
mag_probe
mag_probe/r
mag_probe/r/b
mag_probe/r/bdot
mag_probe/squid
mag_probe/squid/b
mag_probe/squid/bdot
mag_probe/t
mag_probe/t/b
mag_probe/t/bdot
mag_probe/z
mag_probe/z/b
mag_probe/z/bdot
probe_info
probe_info/current_probe_factor
probe_info/probe_hole_separation
probe_info/probe_stalk_diameter
probe_info/rloop_area
probe_info/tloop_area
probe_info/voltage_probe_factor
probe_info/zloop_area
run_info
run_info/Collection Dates
run_info/Discharge_Voltage (kV)
run_info/Gas Close (ms)
run_info/Gas Open (ms)
run_info/Stuffing Delay (ms)
time
time/numsamples
time/time_s
time/time_us
time/tinterval
time/tstart


You should see a long list of names that look like directory names. An HDF5 file works sort of like a filing system where arrays are stored in groups. This kind of system is called a dictionary. Within this dictionary there are two types of items: groups and datasets. Both are named using words in quotations. Groups are like the file folders and datasets are like the files---they are either numbers, strings, or arrays.

To exact a particular dataset from the file, you need to list out the group names and dataset names. For example, there is a dataset that contains magnetic field values in the z-direction (based on cylindrical coordinates) as a function of time for three probes and for 10 shots. To isolate this dataset, we use the following code:

In [3]:
data_br = data1["mag_probe"]["r"]["b"]

To determine if we extracted the right object, we can check the array size of ```data_br``` using:

In [4]:
import numpy as np
np.shape(data_br)

(3, 10, 25003)

As you see, it lists three numbers: The first is the number of probes (there are three magnetic probes in this dataset at three different axial locations), number of shots (we took 10 total shots), and number of samples (there are 25003 timesteps recorded for this data).